src/uu/tail/src/tail.rs

```diff
@@ -65,13 +65,18 @@ fn uu_tail(settings: &Settings) -> UResult<()> {
     // Add `path` and `reader` to `files` map if `--follow` is selected.
     for input in &settings.inputs.clone() {
         match input.kind() {
-            InputKind::File(path) if cfg!(not(unix)) || path != &PathBuf::from(text::DEV_STDIN) => {
-                tail_file(settings, &mut printer, input, path, &mut observer, 0)?;
-            }
             // File points to /dev/stdin here
-            InputKind::File(_) | InputKind::Stdin => {
+            InputKind::Stdin => {
+                tail_stdin(settings, &mut printer, input, &mut observer)?;
+            }
+            // and here. Currently, on Linux, it seems that the presence and absence of this
```

Comment:
If the "presence and absence of this makes no difference", then why is the code there? Why is that true at all, the code obviously behaves differently if the given path is exactly `text::DEV_STDIN`, so the comment seems to be incorrect?

Also, while I agree that the refactoring makes the code nicer, does it actually change any of the functionality? I struggle to find an example where this changes the behavior. Can you explain it to me?

Reply:
I just had a brainfart when I wrote that. Removing it didn't make a change in the tests (so I'll go ahead and add some tests for that tomorrow in addition to the tests for the pid), but leaving it _does_ make it treat `/dev/stdin` as `stdin` on unix systems, which seems to be the proper behavior.

For the refactoring, I _do_ think there is one bug fixed, where on a non-unix system, even if using stdin on that OS, it would default to treating it as a file, as the file match is either (not unix) or `/dev/stdin`. However, I don't dev on Windows, so I haven't tested it yet, so that's just eyeballing it. Outside of just making it more readable, but I'm okay with changing it back into 2 match cases. I don't think it would be too hard to place (unix && stdin) in the stdin arm instead of (!unix || !stdin) in the file arm.

But I'll go ahead and add those tests tomorrow, and I can undo some of the refactor if you want and just fix up that single bug (if it is bug), or leave it as is

Reply:
No need to revert the match-rewrite, I like the new version, because it's clearer what it does. I just don't think there is a difference in behavior, that's why I was surprised by the comment :)